pumps/mongo.go

```diff
@@ -103,6 +115,89 @@ func (m *MongoPump) Init(config interface{}) error {
 	return nil
 }
 
+func (m *MongoPump) capCollection() (ok bool) {
+
+	var colName = m.dbConf.CollectionName
+	var colCapMaxSizeBytes = m.dbConf.CollectionCapMaxSizeBytes
+	var colCapEnable = m.dbConf.CollectionCapEnable
+
+	if !colCapEnable {
+		return false
+	}
+
+	exists, err := m.collectionExists(colName)
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"prefix": mongoPrefix,
+		}).Errorf("Unable to determine if collection (%s) exists. Not capping collection: %s\n", colName, err.Error())
+
+		return false
+	}
+
+	if exists {
+		log.WithFields(logrus.Fields{
+			"prefix": mongoPrefix,
+		}).Warnf("Collection (%s) already exists. Capping could result in data loss. Ignoring\n", colName)
+
+		return false
+	}
+
+	if strconv.IntSize < 64 {
+		log.WithFields(logrus.Fields{
+			"prefix": mongoPrefix,
+		}).Warn("Pump running < 64bit architecture. Not capping collection as max size would be 2gb\n")
+
+		return false
+	}
+
+	if colCapMaxSizeBytes == 0 {
+		defaultBytes := 5
+		colCapMaxSizeBytes = defaultBytes * GiB
+
+		log.WithFields(logrus.Fields{
+			"prefix": mongoPrefix,
+		}).Infof("-- No max collection size set for %s, defaulting to %d", colName, colCapMaxSizeBytes)
+	}
+
+	err = m.dbSession.DB("").C(colName).With(m.dbSession.Copy()).Create(&mgo.CollectionInfo{Capped: true, MaxBytes: colCapMaxSizeBytes})
```

Comment:
I would say it is too smart :)
Personally, I would prefer splitting it into 2 lines, creating a separate session object and using it.